index.js

```diff
@@ -302,7 +302,11 @@ function sanitizeHtml(html, options, _recursing) {
     }
     classes = classes.split(/\s+/);
     return classes.filter(function(clss) {
-      return allowed.indexOf(clss) !== -1;
+      return allowed.some(function(_allowed) {
+        var allowedRegex = new RegExp('^' + _allowed.split('*').join('.+') + '$');
```

Comment:
Apologies for sitting on this for a criminally long period of time!

There does still appear to be an issue, the wildcard is being set up properly but the actual text to be matched is not being escaped properly for use in a regular expression. Something like the regexp-quote module needs to be used to finish this out.